### PR TITLE
[Dashboard] Update the Ray dashboard documentation to explain memory view.

### PR DESCRIPTION
**Pull request by rkooo567**

### CODE REVIEW

File: doc/source/ray-dashboard.rst [hunk 1]
```diff
@@ -56,6 +56,16 @@ The logical view shows you:
 .. image:: https://raw.githubusercontent.com/ray-project/Images/master/docs/dashboard/Logical-view-basic.png
     :align: center
 
+Memory View
+~~~~~~~~~~~~
+The Memory view shows you:
```
Comment (edoakes) [inline doc/source/ray-dashboard.rst:61]:
```suggestion
The memory view shows you:
```

File: doc/source/ray-dashboard.rst [hunk 2]
```diff
@@ -56,6 +56,16 @@ The logical view shows you:
...
+
+- States of Ray objects, such as sizes, refernece types, and call sites.
```
Comment (edoakes) [inline doc/source/ray-dashboard.rst:63]:
```suggestion
- The state of Ray objects, including their size, reference type, and call site.
```

File: doc/source/ray-dashboard.rst [hunk 3]
```diff
@@ -56,6 +56,16 @@ The logical view shows you:
...
+- Summary of reference types and object sizes in use.
```
Comment (edoakes) [inline doc/source/ray-dashboard.rst:64]:
```suggestion
- A summary of reference types and object sizes in use.
```

File: doc/source/ray-dashboard.rst [hunk 4]
```diff
@@ -124,6 +134,11 @@ As a result, the rest of ``Actor1`` will be pending.
 You can also see it is infeasible to create ``Actor2`` because it requires 4 GPUs which 
 is bigger than the total gpus available in this cluster (2 GPUs). 
 
+Debugging ObjectStoreFullError and Memory Leak
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+You can get information for Ray objects through the Ray Memory tab. It is useful to debug memory leak, especially the `ObjectStoreFullError` error.
```
Comment (edoakes) [inline doc/source/ray-dashboard.rst:139]:
```suggestion
You can view information for Ray objects in the memory tab. It is useful to debug memory leaks, especially `ObjectStoreFullError`.
```

File: doc/source/ray-dashboard.rst [hunk 5]
```diff
@@ -124,6 +134,11 @@ As a result, the rest of ``Actor1`` will be pending.
...
+Note that this is the ported version of the `ray memory command <https://docs.ray.io/en/latest/memory-management.html#debugging-using-ray-memory>`_. To learn how to translate this table, please learn the `ray memory` command.
```
Comment (edoakes) [inline doc/source/ray-dashboard.rst:140]:
```suggestion
Note that this is the same information as displayed in the `ray memory command <https://docs.ray.io/en/latest/memory-management.html#debugging-using-ray-memory>`_. For details about the information contained in the table, please see the `ray memory` documentation.
```

**Review by edoakes — APPROVED** [review]

**Comment by rkooo567** [comment]
@edoakes All comments are addressed!

**Pull request merged**